tests/anonymous-credentials/schema.spec.ts

```diff
@@ -13,17 +13,118 @@ import {
   beforeAll(async () => {
     await initializeWasm();
   });
 
+  it('Parse JSON-schema syntax', () => {
+    const jsonSchema = {
+      type: 'object',
+      properties: {
+        credentialVersion: {type: 'string'},
+        credentialSchema: {type: 'string'},
+        credentialSubject: {
+          type: 'object',
+          properties: {
+            SSN: { $ref: '#/definitions/encryptableString' },
+            userId: { $ref: '#/definitions/encryptableCompString' },
+            vision: {type: 'integer', minimum: -20},
+            longitude: {type: 'number', minimum: -180, multipleOf: 0.001},
+            time: { $ref: '#/definitions/positiveInteger' },
+            weight: {
+              allOf: [
+                { $ref: '#/definitions/positiveNumber' },
+                { multipleOf: 0.1 }
+              ]
+            },
+          }
+        }
+      }
+    };
+    const schema = CredentialSchema.convertToInternalSchemaObj(jsonSchema);
+    expect(schema).toEqual({
+      credentialVersion: { type: 'string' },
+      credentialSchema: { type: 'string' },
+      credentialSubject: {
+        SSN: { type: 'stringReversible', compress: false },
+        userId: { type: 'stringReversible', compress: true },
+        vision: { type: 'integer', minimum: -20 },
+        longitude: { type: 'decimalNumber', minimum: -180, decimalPlaces: 3 },
+        time: { type: 'positiveInteger' },
+        weight: { type: 'positiveDecimalNumber', decimalPlaces: 1 }
+      }
+    });
+
+    const jsonSchema1 = {
+      type: 'object',
+      properties: {
+        credentialVersion: { type: 'string' },
+        credentialSchema: { type: 'string' },
+        credentialSubject: {
+          type: 'object',
+          properties: {
+            sensitive: {
+              type: 'object',
+              properties: {
+                SSN: { $ref: '#/definitions/encryptableString' },
+                userId: { $ref: '#/definitions/encryptableCompString' },
+              }
+            },
+            longitude: { type: 'number', minimum: -180, multipleOf: 0.001 },
+            time: { $ref: '#/definitions/positiveInteger' },
+            physical: {
+              type: 'object',
+              properties: {
+                l1: {
+                  type: 'object',
+                  properties: {
+                    vision: { type: 'integer', minimum: -20 },
+                  }
+                },
+                l2: {
+                  type: 'object',
+                  properties: {
+                    weight: {
+                      allOf: [
+                        { $ref: '#/definitions/positiveNumber' },
+                        { multipleOf: 0.1 }
+                      ]
+                    },
+                  },
+                }
+              }
+            }
+          },
+        }
+      }
+    };
+    const schema1 = CredentialSchema.convertToInternalSchemaObj(jsonSchema1);
+    expect(schema1).toEqual({
```

Comment:
don't really see the need to convert it to another object like this if it worked by just flattening before. all this has done is remove my work and had yourself do duplicate work essentially in parsing it etc

Reply:
Benefits of converting 
1. It makes the implementation more efficient. Previously whenever schema needed to be flattened (in `flattenSchemaObj`), `flattenJSONSchema` would be called to convert the JSON-schema "style" object to another object. And schema flattening is done in both credentials and presentations. With my approach, the conversion from JSON-schema format to the one used internally is done only once while creating the `CredentialSchema` object.
2. By doing the conversion while creating `CredentialSchema` object, very little of the previous code has to change as that logic is oblivious to the fact that JSON-schema format is being used as input.

> had yourself do duplicate work essentially in parsing it etc

Much of that work is in tests which is essential. Just compare the number of checks done in your PR vs mine.

Reply:
we touched on this on the call, but my initial understanding was that flattening was only called once in the constructor anyway. but i didnt look deeply into that. its not a big deal, however i disliked accessing it from outside this class as a library caller may think that mutating that value will modify the behavior/actual schema. perhaps making it private works